[admin]/admin/server/admin_ip2c.lua

```diff
@@ -1,352 +1,13 @@
-	-- filename is file to read
-	new = function(self, filename)
-		local obj = setmetatable({}, { __index = FileLineReader })
-		self.hFile = fileOpen( filename )
-		self.buffer = ""
-		return obj
-	end,
-
-	-- Close file
-	close = function(self)
-		if self.hFile then
-			fileClose( self.hFile )
-		end
-		self.hFile = nil
-	end,
-
-	-- Read line. Return false if EOF
-	readLine = function(self)
-		if not self.hFile then return false end
-		while true do
-			local endpos = string.find(self.buffer, "\n")
-			-- Found '\n' ?
-			if endpos then
-				local line = string.sub(self.buffer, 1, endpos - 1)
-				self.buffer = string.sub(self.buffer, endpos + 1)
-				return line
-			end
-			-- Get more bytes if possible
-			if fileIsEOF( self.hFile ) then
-				if string.len( self.buffer ) > 0 then
-					-- Last line has no '\n'
-					local line = self.buffer
-					self.buffer = ""
-					return line
-				end
-				self:close()
-				return false
-			end
-			self.buffer = self.buffer .. fileRead( self.hFile, 500 )
-		end
-	end,
-}
-
-----------------------------------------------------------------------------------------
--- CoroutineSleeper
---   Wrapper for coroutine which can sleep and automatically resume
-----------------------------------------------------------------------------------------
-CoroutineSleeper = {
-	-- myFunc is coroutine entry point
-	new = function(self, myFunc, ...)
-		local obj = setmetatable({}, { __index = CoroutineSleeper })
-		-- Use inner function to call myFunc, so we can auto :detach when finished
-		obj.handle = coroutine.create( function(obj2, ...)
-											myFunc(obj2, ...)
-											obj:detach()
-										end )
-		coroutine.resume(obj.handle,obj, ...)
-		return obj
-	end,
-
-	-- Remove ref to coroutine
-	detach = function(self)
-		self.handle = nil
-	end,
-
-	-- Check if still has ref to coroutine
-	isAttached = function(self)
-		return self.handle ~= nil
-	end,
-
-	-- Sleep for a bit, then automatically resume
-	sleep = function(self, ms)
-		if not self:isAttached() then return end
-		setTimer( function()
-			if not self:isAttached() then return end
-			local status = coroutine.status(self.handle)
-			if (status == "suspended") then
-				coroutine.resume(self.handle)
-			elseif (status == "dead") then
-				self.handle = nil
-			end
-		end, math.max( ms, 50 ), 1 )
-		coroutine.yield()
-	end,
-}
+--[[
+    ip2c script has been moved to its own separate resource
+
+    Placeholder function to preserve backwards compatibility
+        e.g. custom external scripts that call this function
+]]
+function getPlayerCountry(player)
+    outputServerLog("[Deprecation Warning] admin/server/admin_ip2c.lua: getPlayerCountry(player)")
+    outputServerLog("  This function has been moved to the new ip2c standalone resource")
+    outputServerLog("  Get it from the latest MTA:SA default resources package:")
+    outputServerLog("  https://mirror-cdn.multitheftauto.com/mtasa/resources/mtasa-resources-latest.zip", 2)
+    return false
```

Comment:
```suggestion
    local ip2c = getResourceFromName('ip2c')
    return ip2c and call(ip2c, 'getPlayerCountry', player)
```

Instead of just returning false, why not forward the call to the ip2c resource? This way if you have ip2c resource and your resources call ip2c from admin you get the deprecation warning but your scripts will not break

Reply:
Done @brenodanyel